### ISSUE TITLE
--agentless: agent not removed in case of error

### ISSUE THREAD

**Issue by multi-io** [aylei/kubectl-debug#57]
While investigating https://github.com/aylei/kubectl-debug/issues/52 I saw this:

```
$ kubectl -n default get pod
No resources found.
$ kubectl debug harbor-harbor-portal-68df4cdb58-9xx4l --agentless --port-forward
Agent Pod info: [Name:debug-agent-pod-853aee9e-db9b-11e9-b976-88e9fe6345ac, Namespace:default, Image:aylei/debug-agent:latest, HostPort:10027, ContainerPort:10027]
Waiting for pod debug-agent-pod-853aee9e-db9b-11e9-b976-88e9fe6345ac to run...
error: container [portal] not ready
$ kubectl -n default get pod
NAME                                                   READY   STATUS    RESTARTS   AGE
debug-agent-pod-853aee9e-db9b-11e9-b976-88e9fe6345ac   1/1     Running   0          30s
$
```

That should've been deleted by now, right?

**Comment by aylei** [comment]
Yes, that container should have been deleted.
@tkanng could you please take a look?

**Comment by calmkart** [comment]
i cant reproduce this bug
when i run this command, i get

`
Error from server (NotFound): pods "harbor-harbor-portal-68df4cdb58-9xx4l" not found
`

this code can get the pod, if not found it will return 
https://github.com/aylei/kubectl-debug/blob/3b7576c2194e36a12eddd434d2a3fb13042f9c60/pkg/plugin/cmd.go#L321-L326

 i guess this bug maybe about your kubectl-debug version or kubernetes version

**Comment by calmkart** [comment]
like [52](https://github.com/aylei/kubectl-debug/issues/52)
its happends in readness Probe not ready pod, completed pod or faild pod.

if i run a computes pi job and completed, then i start kubectl-debug for that completed pod,i'll see
`error: cannot debug in a completed pod; current phase is Succeeded`

and the pod is also not deleted
```
[root@k8s-master ~]# kubectl get pods    
NAME                                                   READY   STATUS      RESTARTS   AGE 
debug-agent-pod-49f36a30-e048-11e9-9b1e-784f439aad34   1/1     Running     0          5m26s    
pi-gw8s7                                               0/1     Completed   0          12m     
```

i think the issue happends around this block of code
https://github.com/aylei/kubectl-debug/blob/3b7576c2194e36a12eddd434d2a3fb13042f9c60/pkg/plugin/cmd.go#L363-L370

though return error,but not delete agent pod.maybe others situation can also Trigger this bug.
i'm trying to fix it, probably few days.

**Comment by calmkart** [comment]
> like [52](https://github.com/aylei/kubectl-debug/issues/52)
> its happends in readiness Probe not ready pod, completed pod or faild pod and so on.
> 
> if i run a computes pi job and completed, then i start kubectl-debug for that completed pod,i'll see
> `error: cannot debug in a completed pod; current phase is Succeeded`
> 
> and the pod is also not deleted
> 
> ```
> [root@k8s-master ~]# kubectl get pods    
> NAME                                                   READY   STATUS      RESTARTS   AGE 
> debug-agent-pod-49f36a30-e048-11e9-9b1e-784f439aad34   1/1     Running     0          5m26s    
> pi-gw8s7                                               0/1     Completed   0          12m     
> ```
> 
> i think the issue happends around this block of code
> https://github.com/aylei/kubectl-debug/blob/3b7576c2194e36a12eddd434d2a3fb13042f9c60/pkg/plugin/cmd.go#L363-L370
> 
> though return error,but not delete agent pod.maybe others situation can also Trigger this bug.
> i'm trying to fix it, probably few days.

fixed in [pull request#63](https://github.com/aylei/kubectl-debug/pull/63)
